inference-engine/src/gna_plugin/transformations/split_convolution_with_large_buffer_size.cpp

```diff
@@ -0,0 +1,105 @@
+
+using namespace GNAPluginNS;
+
+NGRAPH_RTTI_DEFINITION(SplitConvolution, "SplitConvolution", 0);
+
+static std::vector<int64_t> GetConvSplitSizes(std::shared_ptr<ngraph::Node> conv) {
+    uint32_t width = conv->get_input_shape(0).back();
+    uint32_t in_channels = conv->get_input_shape(0).at(1);
+    uint32_t usedWidth = 0;
+    std::vector<int64_t> split_sizes;
+    uint32_t width_max_size = GNALimitations::bufferMaxSize / in_channels;
+    width_max_size = width_max_size - width_max_size % 64;
+    while (usedWidth < width) {
+        uint32_t width_part = std::min(width - usedWidth, width_max_size);
+        split_sizes.push_back(width_part);
+        usedWidth += width_part;
+    }
+    IE_ASSERT(usedWidth == width);
+    return split_sizes;
+}
+
+SplitConvolution::SplitConvolution() {
+    auto conv = ngraph::pattern::wrap_type<ngraph::opset7::Convolution>({ngraph::pattern::any_input(),
+        ngraph::pattern::any_input()});
+    auto add = ngraph::pattern::wrap_type<ngraph::opset7::Add>({conv,
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>()});
+    auto conv_output = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{conv, add});
+    auto out_fq = ngraph::pattern::wrap_type<ngraph::opset7::FakeQuantize>({conv_output,
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset7::Constant>()});
+    auto root = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{conv_output, out_fq});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto& pattern_map = m.get_pattern_value_map();
+        auto conv_node = pattern_map.at(conv).get_node_shared_ptr();
+        auto input_size = std::accumulate(std::begin(conv_node->get_input_shape(0)),
+            std::end(conv_node->get_input_shape(0)), 1, std::multiplies<size_t>());
+        if (input_size <= GNALimitations::bufferMaxSize) {
+            return false;
+        }
+
+        auto split_sizes = GetConvSplitSizes(conv_node);
+        IE_ASSERT(split_sizes.size() > 1);
+
+        auto conv_output = conv_node->output(0).get_target_inputs().begin()->get_node()->shared_from_this();
+        std::shared_ptr<ngraph::Node> bias_node = nullptr;
+        if (std::dynamic_pointer_cast<ngraph::opset7::Add>(conv_output)) {
+            auto add_second_input = conv_output->input_value(1).get_node_shared_ptr();
+            if (std::dynamic_pointer_cast<ngraph::opset7::Constant>(conv_output)) {
+                bias_node = conv_output->input_value(1).get_node_shared_ptr();
+                conv_output = conv_output->output(0).get_target_inputs().begin()->get_node()->shared_from_this();
+            }
+        }
+        std::shared_ptr<ngraph::Node> fq_node = std::dynamic_pointer_cast<ngraph::opset7::FakeQuantize>(conv_output);
```

Comment:
The same question. Why you don't use `pattern_map` to find either node exists or not?

Reply:
usedpattern_map